presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2050,6 +2055,30 @@ private PhysicalOperation createLookupJoin(
             createDynamicFilter(node, context, partitionCount).ifPresent(
                     filter -> factoriesBuilder.add(createDynamicFilterSourceOperatorFactory(filter, node, buildSource, buildContext)));
 
+            if (isEnableDynamicFiltering(context.getSession()) && node.getType() == INNER && node.getCriteria().size() > 0) {
+                List<DynamicFilterSourceOperator.Channel> filterBuildChannels = node.getDynamicFilters()
+                        .entrySet()
+                        .stream()
+                        .map(entry -> {
+                            String filterId = entry.getKey();
+                            Symbol buildSymbol = entry.getValue();
+                            int buildChannelIndex = buildNode.getOutputSymbols().indexOf(buildSymbol);
+                            verify(buildChannelIndex >= 0);
+                            Type type = buildSource.getTypes().get(buildChannelIndex);
+                            return new DynamicFilterSourceOperator.Channel(filterId, type, buildChannelIndex);
+                        })
+                        .collect(toImmutableList());
+
+                DynamicFilterSourceConsumer dynamicFilterSourceConsumer = context.createDynamicFiltersConsumer(node.getId().toString(), partitionCount);
+                factoriesBuilder.add(new DynamicFilterSourceOperator.DynamicFilterSourceOperatorFactory(
```

Comment:
this will create redundant `DynamicFilterSourceOperatorFactory` with respect to local dynamic filtering.

Reply:
Made changes to reuse same DynamicFilterSourceOperatorFactory for both local and global dynamic filtering